### PR TITLE
QA Framework

### PR DESCRIPTION
**Pull request by DominicMonares**
This PR implements the following:
- QA (primary component)
- 4 child components: QAnswer, QASearch, QAsk, and QAView
- Server route for Q&A component
- API call made from server (successfully getting data all the way back to client)

I added separators in the server and apiCalls files just to keep things a bit more organized. Please let me know if y'all want me to get rid of/change these.

Lastly, I couldn't get ajaxPrefilter to work for the life of me, so I changed my config.js and the example file to match it. Please let me know if y'all are doing something similar/different, we might want to discuss configs when we next meet just to make sure we're all on the same page with them. This way, we can set up a single API_KEY variable in the apiCalls file that everyone can use.

Please let me know what y'all think and if there are any changes you'd like me to make.

### CODE REVIEW

**Review by RobStafford — CHANGES_REQUESTED** [review]
Same comment here. HR/SEIR suggests "react-dev": "webpack --watch", is preferable to "react-dev": "webpack --config webpack.config.js -w", in package.json, Otherwise, no issues that I see.

**Comment by RobStafford** [comment]
&, as I said in slack, if we believe strongly we know why the checks are failing & don't think it will be a problem, I'm still ok turning my key.

File: server/index.js [hunk 1]
```diff
@@ -4,16 +4,30 @@ const path = require('path');
...
+// ========== Shared ========== //
+
+// ========== Overview ========== //
+
+// ========== Related Products ========== //
+
+// ========== Questions & Answers ========== //
+
+app.get('/qa/questions/:product_id', function(req, res) {
+  const apiCalls = require('../utils/apiCalls.js');
```
Comment (dwightdoran) [inline server/index.js]: Just a suggestion, but maybe we could put that require on line 20 outside of the scope of that function so we all can pull from the same variable as apposed to re-declaring the variable.

File: server/index.js [hunk 2]
```diff
@@ -4,16 +4,30 @@ const path = require('path');
...
+  let product = req.params.product_id;
+
+  apiCalls.getProductQuestionData(product, (questions) => {
+    res.send(questions);
+  })
+})
+
+// ========== Ratings & Reviews ========== //
+
+// ========== Server Listen ========== //
```
Comment (dwightdoran) [inline server/index.js:32]: I'm a really big fan of the separation comments for organization. +1

**Review by dwightdoran — APPROVED** [review]
I approve. Just a minor comment on where that variable was declared.

File: server/index.js (hunk 1 above)
Comment (DominicMonares) [reply server/index.js]: Good call, moved the declaration to top of doc

**Comment by DominicMonares** [comment]
Thank you for the reviews y'all! All changes have been made and merge conflicts resolved.

**Comment by DominicMonares** [comment]
FYI - I just changed the index.jsx state to a hardcoded product id so we can keep our components dynamic, as Rob suggested on Dwight's PR.

**Pull request merged**